### PR TITLE
swupdate: systemd: Fix missing type in service section  

### PR DESCRIPTION
**Pull request by jorisoffouga**
Signed-off-by: Joris Offouga <offougajoris@gmail.com>

### CODE REVIEW

File: recipes-support/swupdate/swupdate/swupdate.service [hunk 1]
```diff
@@ -2,9 +2,10 @@
 Description=SWUpdate daemon
 Documentation=https://github.com/sbabic/swupdate
 Documentation=https://sbabic.github.io/swupdate
-After=swu-environment.service
+After=swu-environment.service swupdate.socket
 
 [Service]
+Type=notify
```
Comment (texierp) [inline recipes-support/swupdate/swupdate/swupdate.service:8]: Indeed, good catch !

File: recipes-support/swupdate/swupdate/swupdate.service [hunk 2]
```diff
@@ -2,9 +2,10 @@
 Description=SWUpdate daemon
 Documentation=https://github.com/sbabic/swupdate
 Documentation=https://sbabic.github.io/swupdate
-After=swu-environment.service
+After=swu-environment.service swupdate.socket
```
Comment (texierp) [inline recipes-support/swupdate/swupdate/swupdate.service]: swupdate.socket is required only if we rely on socket-based 
activation (not the case in this layer).
In fact, I would prefer to keep a "minimal" configuration here, 
and let the final user decide.

File: recipes-support/swupdate/swupdate/swupdate.socket [hunk 3]
```diff
@@ -0,0 +1,11 @@
+[Unit]
```
Comment (texierp) [inline recipes-support/swupdate/swupdate/swupdate.socket]: see above

File: recipes-support/swupdate/swupdate_%.bbappend [hunk 4]
```diff
@@ -7,6 +7,9 @@ SRC_URI += " \
 "
 
 do_install_append() {
-    install -d ${D}${sysconfdir}
-    install -m 644 ${WORKDIR}/swupdate.cfg ${D}${sysconfdir}
+    install -Dm 644 ${WORKDIR}/swupdate.cfg ${D}${sysconfdir}
+    install -m 644 ${WORKDIR}/swupdate.socket ${D}${systemd_unitdir}/system
```
Comment (texierp) [inline recipes-support/swupdate/swupdate_%.bbappend]: see above

File: recipes-support/swupdate/swupdate_%.bbappend [hunk 5]
```diff
@@ -7,6 +7,9 @@ SRC_URI += " \
...
 }
+
+FILES_${PN} += "${systemd_unitdir}/system/swupdate.socket"
+SYSTEMD_SERVICE_${PN} += "swupdate.socket"
```
Comment (texierp) [inline recipes-support/swupdate/swupdate_%.bbappend]: see above

**Review by texierp — CHANGES_REQUESTED** [review]

File: recipes-support/swupdate/swupdate/swupdate.service (hunk 2 above)
Comment (jorisoffouga) [reply recipes-support/swupdate/swupdate/swupdate.service]: Ah okay, so i remove the install socket and just add swupdate.socket ?

Comment (texierp) [reply recipes-support/swupdate/swupdate/swupdate.service]: Just drop the stuff related to swupdate.socket and keep
only the fix about Type=notify.

So, please rebase with a new commit log which explain
why we have to set this specific Type (see https://www.freedesktop.org/software/systemd/man/systemd.service.html)

**Comment by texierp** [comment]
LGTM

**Review by texierp — APPROVED** [review]

**Pull request merged**